clients/githubrepo/branches.go

```diff
@@ -481,8 +482,9 @@ nextRule:
 			}
 		}
 
+		targetsBranches := rule.Target != nil && *rule.Target == "branch"
```

Comment:
is this lower case for a reason? When I check one of my test repos I see it in all caps.
```
"target": "BRANCH",
```

If it's inconsistent, or we want to be more defensive, we use `strings.EqualFold` in other places for case insensitive comparisons.